### PR TITLE
Migrated test_peer_probe_firewall_ports_not_opened

### PR DESCRIPTION
**Pull request by neesingh-rh**
Migrated test_peer_probe_firewall_ports_not_opened.

Fixes: #429

### CODE REVIEW

File: tests/functional/glusterd/test_profile_operations_with_one_node_down.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
+#  Copyright (C) 2019-2020  Red Hat, Inc. <http://www.redhat.com>
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_profile_operations_with_one_node_down.py:1]: @neesingh-rh why are we taking this test file...Is this going to be taken along with the above test case in this patch itself ?

Comment (neesingh-rh) [reply tests/functional/glusterd/test_profile_operations_with_one_node_down.py:1]: No,it was by mistake

**Comment by neesingh-rh** [comment]
test_peer_probe_firewall_ports_not_opened-Generic PASS

Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_peer_probe_firewall_ports_not_opened
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|   Generic   |     PASS    | 21.468557834625244 |
+-------------+-------------+--------------------+



Framework runtime : 26.572521209716797

**Comment by neesingh-rh** [comment]
x-special/nautilus-clipboard
copy
file:///home/neerajpsingh/Pictures/peer_probe_firewall.png

**Comment by neesingh-rh** [comment]
image

File: tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        2. Perform peer probe to Node2 from Node 1
+        3. Check for core files created
+        """
+        # Timestamp of current test case of start time
+        ret = redant.execute_abstract_op_node('date +%s', self.server_list[0])
+        test_timestamp = ret['msg'][0].rstrip('\n')
+
+        # Remove firewall service on the node to probe
+        self.node = choice(self.server_list[1:])
+        for server in self.server_list[1:]:
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py:62]: we can use the destroy_cluster function here instead. This logic is handled inside it.

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            redant.peer_detach(server, self.server_list[0])
+        for server in self.server_list:
+            cmd = "systemctl restart firewalld"
+            redant.execute_abstract_op_node(cmd, server)
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py:66]: There isn't any restart in the glusto TC, any specific reasons for adding this?

File: tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        self._remove_firewall_service(redant)
+
+        # Try peer probe from one node to another
+        try:
+            redant.peer_probe(self.node, self.server_list[0])
+        except:
+            msg = (f"Peer probe of {self.node} from"
+                   f"{self.server_list[0]} failed as expected ")
+            redant.logger.info(msg)
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py:75]: Also, you need to verify the `error_msg` from the command execution as per the glusto TC. You can use the `excep` arg to pass `False` to it and get the complete ret value from the command execution.

File: tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+        # Verify no core files are created
+        ret = redant.check_core_file_exists(self.server_list, test_timestamp)
+        if ret:
+            raise Exception("Core file is found")
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py:80]: The TC asks to check `ERROR` logs in the `glusterd.log` file before checking for the core file. Please update.

File: tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+        self._add_firewall_service(redant)
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_peer_probe_firewall_ports_not_opened.py:82]: This call and function can be removed and instead use the code in the terminate function, as mentioned above.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

**Comment by nik-redhat** [comment]
Closing this PR, I will send out a separate PR for the TC.

**Pull request closed without merging**